core/state/statedb.go

```diff
@@ -288,15 +317,14 @@ func (self *StateDB) newStateObject(addr common.Address) *StateObject {
 	if glog.V(logger.Core) {
 		glog.Infof("(+) %x\n", addr)
 	}
-
-	stateObject := NewStateObject(addr, self.db)
-	stateObject.SetNonce(StartingNonce)
-	self.stateObjects[addr.Str()] = stateObject
-
-	return stateObject
+	obj := NewObject(addr, Account{})
+	obj.dirty = true
+	obj.SetNonce(StartingNonce)
+	self.stateObjects[addr] = obj
+	return obj
 }
 
-// Creates creates a new state object and takes ownership. This is different from "NewStateObject"
+// Creates creates a new state object and takes ownership.
```

Comment:
I'd just like to verify; the `CreateStateObject` does not stash away the `data` into `s.all`, so it does not wind up in there until after `commit`. Is that intended behaviour?